src/RollOfHonour.Data/Repositories/MemorialRepository.cs

```diff
@@ -81,24 +81,25 @@ public async Task<IEnumerable<Memorial>> GetAll()
         }
     }
 
-    public async Task<IEnumerable<MemorialSearchResult>?> FindMemorialByName(string nameFragment)
+    public async Task<PaginatedList<Core.Models.Memorial>> SearchMemorials(string searchString, int pageIndex, int pageSize)
     {
         var dbMemorials = _dbContext.WarMemorials.Where(m =>
-            m.Name.Contains(nameFragment))
-            .AsNoTracking();
+            m.Name.Contains(searchString));
 
-        if (!dbMemorials.Any())
+        var resultCount = dbMemorials.Count();
+
+        if (resultCount == 0)
         {
-            return null;
+            //TODO: handle case where there are none
+            throw new NotImplementedException();

```

Comment:
Even an empty PageList would feel better than this.

Reply:
For now, this will error, I will resolve in the next PR so that we unblock your image piece